cpp/support/thread_safe_cache.h

```diff
@@ -179,6 +186,215 @@ class ThreadSafeCache<Key, Value> {
   std::shared_mutex erase_mutex_;
 };
 
+namespace details {
+
+template <typename Key, typename Value>
+class LRUCacheImpl {
+ protected:
+  struct Entry {
+    Value value;  // value of the node
+    int index;    // node index
+  };
+
+  /*! \brief Visits the node and moves it to the back of the LRU list. Return its value. */
+  auto LRUvisit(const std::pair<const Key, Entry>& pair) -> const Value& {
```

Comment:
```suggestion
  auto LRUVisit(const std::pair<const Key, Entry>& pair) -> const Value& {
```
plz check this file again for style